### DISCUSSION TITLE
MARKER_RIEMANN Boundary Problem

### DISCUSSION THREAD

**Discussion by safdarianebi** [su2code/SU2#2306] [Q&A]
Hello!

I have a simple question about the Riemann type for the boundary condition. I would use the type "TOTAL_SUPERSONIC_INFLOW" for the MARKER_RIEMANN but the warning "Invalid Riemann input!" comes out.

I have controlled in the SU2/Common/include/option_structure.hpp and this option is present.

How could I do to use it?

this is the boundary defined:
`
MARKER_RIEMANN= (stage1_inflow, TOTAL_SUPERSONIC_INFLOW  , 101325.0, 288.15, 400.0, 0.0, 0.0)`

and this is the error:

```
------------------------------ Begin Solver -----------------------------

Simulation Run using the Multizone Driver

 Setting translational grid velocities.
 Non-dim. translational velocity: (0, 296.633, 0).
WARNING: SU2 was not compiled for an AVX-capable architecture. Performance could be better,
         see https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/#compiler-optimizations


Error in "virtual void CEulerSolver::BC_TurboRiemann(CGeometry*, CSolver**, CNumerics*, CNumerics*, CConfig*, short unsigned int)": 
-------------------------------------------------------------------------
Invalid Riemann input!
------------------------------ Error Exit -------------------------------
```

Thanks!

**Comment by bigfooted** [comment]
It looks like the keyword is still there but the implementation was removed. We have STATIC_SUPERSONIC_INFLOW_PT and STATIC_SUPERSONIC_INFLOW_PD that are probably what you need to use for your testcase.

**Reply by safdarianebi** [reply]
I also tried these options and I got the same error. When I only use MARKER_GILES like this:

`MARKER_GILES= ( stage1_inflow, TOTAL_CONDITIONS_PT, 101325.0, 288.15, 1.0, 0.0, 0.0,1.0,1.0, stage1_outmix1, MIXING_OUT, 0.0, 0.0, 0.0, 0.0, 0.0,1.0,1.0, stage1_inmix1, MIXING_IN, 0.0, 0.0, 0.0, 0.0, 0.0,1.0, 1.0, stage1_outflow, STATIC_PRESSURE, 125000.0, 0.0, 0.0, 0.0, 0.0,1.0,1.0 )`

the solution doesn't converge or the result is not satisfactory because the inlet Mach number, which is supersonic, is lower than one, and this is why I tried Riemann Boundary.

My case is 2-D and inflow Mach number is supersonic. After many difficulties to fix this I am apprehensive that I can not use SU2 for my case.

Additionally, my mesh and geometry can be seen (it is a blade-to-blade view of a transonic compressor):

![Screenshot (1428)](https://github.com/su2code/SU2/assets/125588074/4c724338-425b-442d-9d83-1aa3e7a80558)

**Reply by bigfooted** [reply]
I see that you are trying to use the turbomachinery setup, for bc_turbo_riemann we have TOTAL_CONDITIONS_PT, MIXING_IN, MIXING_OUT, STATIC_PRESSURE and RADIAL_EQUILIBRIUM as boundary conditions. 
Maybe you can first try one of the testcases in the TestCases folder.

**Reply by safdarianebi** [reply]
Thank you for your guidance. The version I am using is **SU2-v8.0.0-turbo-linux64-mpi**. Could you please tell me if su2 can simulate transonic compressors especially when we have supersonic inflow? and which boundaries are suitable for supersonic inflow and do not apply any limit for Mach number?
